File: ChristmasTree_API/Models/Category.cs
namespace ChristmasTreeApp.Models
{
    public class Category
    {
        public int CategoryId { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string Icon { get; set; }
        public ICollection<Product> Products { get; set; }
    }
}

File: ChristmasTreeApp/lib/models/category.dart
class Category {
  final int categoryId;
  final String name;
  final String description;
  final String icon;

  Category({
    required this.categoryId,
    required this.name,
    required this.description,
    required this.icon,
  });

  factory Category.fromJson(Map<String, dynamic> json) {
    return Category(
      categoryId: json['categoryId'],
      name: json['name'],
      description: json['description'],
      icon: json['icon'],
    );
  }
}
